src/gpacmp4/mp4.c

```diff
@@ -359,247 +535,180 @@ int processmp4 (struct lib_ccx_ctx *ctx,struct ccx_s_mp4Cfg *cfg, char *file)
-					mprint ("Submitting this video file will help add support to this case.\n");
+				GF_ISOSample *sample = gf_isom_get_sample(f, i + 1, k + 1, &StreamDescriptionIndex);
+				if (ProcessingStreamDescriptionIndex && ProcessingStreamDescriptionIndex != StreamDescriptionIndex) {
+					mprint("This sample seems to have more than one description. This isn't supported yet.\n");
+					mprint("Submitting this video file will help add support to this case.\n");
 					break;
 				}
 				if (!ProcessingStreamDescriptionIndex)
-						ProcessingStreamDescriptionIndex=StreamDescriptionIndex;
-				if (sample==NULL)
+					ProcessingStreamDescriptionIndex = StreamDescriptionIndex;
+				if (sample == NULL)
 					continue;
-#ifdef DEBUG
-				 mprint ("Data length: %lu\n",sample->dataLength);
-				const LLONG timestamp = (LLONG )((sample->DTS + sample->CTS_Offset) * 1000) / timescale;
+#ifdef MP4_DEBUG
+				mprint("Data length: %lu\n", sample->dataLength);
+				const LLONG timestamp = (LLONG)((sample->DTS + sample->CTS_Offset) * 1000) / timescale;
 #endif
-				set_current_pts(dec_ctx->timing, (sample->DTS + sample->CTS_Offset)*MPEG_CLOCK_FREQ/timescale);
+				set_current_pts(dec_ctx->timing, (sample->DTS + sample->CTS_Offset)*MPEG_CLOCK_FREQ / timescale);
 				set_fts(dec_ctx->timing);
 
 				int atomStart = 0;
-				// process Atom by Atom
-				while (atomStart < sample->dataLength)
-				{
+				// Process Atom by Atom
+				while (atomStart < sample->dataLength) {
 					char *data = sample->data + atomStart;
-					unsigned int atomLength = RB32(data);
-					if (atomLength < 8 || atomLength > sample->dataLength)
-					{
-						mprint ("Invalid atom length. Atom length: %u,  should be: %u\n", atomLength, sample->dataLength);
+					size_t atom_length = -1;
+					switch (track_type) {
+					case MEDIA_TYPE(GF_ISOM_MEDIA_TEXT, GF_ISOM_SUBTYPE_TX3G): // text:tx3g
+					case MEDIA_TYPE(GF_ISOM_MEDIA_SUBT, GF_ISOM_SUBTYPE_TX3G): // sbtl:tx3g (they're the same)
+						atom_length = process_tx3g(ctx, enc_ctx, dec_ctx,
+							&dec_sub, &mp4_ret,
+							data, sample->dataLength, 0);
 						break;
+					case MEDIA_TYPE(GF_ISOM_MEDIA_CAPTIONS, GF_ISOM_SUBTYPE_C608): // clcp:c608
+					case MEDIA_TYPE(GF_ISOM_MEDIA_CAPTIONS, GF_ISOM_SUBTYPE_C708): // clcp:c708
+						atom_length = process_clcp(ctx, enc_ctx, dec_ctx,
+							&dec_sub, &mp4_ret, subtype,
+							data, sample->dataLength);
+						break;
+					default:
+						// See https://gpac.wp.imt.fr/2014/09/04/subtitling-with-gpac/ for more possible types
+						mprint("\nUnsupported track type %c%c%c%c:%c%c%c%c! Please report.\n",
+							(unsigned char)(type >> 24 % 0x100), (unsigned char)((type >> 16) % 0x100),
+							(unsigned char)((type >> 8) % 0x100), (unsigned char)(type % 0x100),
+							(unsigned char)(subtype >> 24 % 0x100), (unsigned char)((subtype >> 16) % 0x100),
+							(unsigned char)((subtype >> 8) % 0x100), (unsigned char)(subtype % 0x100));
+
+						mprint("", track_type);
```

Comment:
track_type is not being consumed by mprint()
It it's a char *, then mprint("%s", track_type)